test/test_irods_http_api.py

```diff
@@ -848,6 +848,65 @@ def test_modifying_permissions_atomically(self):
         self.assertEqual(r.status_code, 200)
         self.assertEqual(r.json()['irods_response']['error_code'], 0)
 
+    def test_modifying_replica_properties(self):
+        headers = {'Authorization': 'Bearer ' + self.rodsadmin_bearer_token}
+
+        # Create a data object.
+        data_object = os.path.join('/', self.zone_name, 'home', self.rodsadmin_username, 'modrepl.txt')
+        r = requests.post(self.url_endpoint, headers=headers, data={
+            'op': 'touch',
+            'lpath': data_object
+        })
+        #print(r.content) # Debug
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(r.json()['irods_response']['error_code'], 0)
+
+        # Show the replica is currently marked as good.
+        r = requests.get(f'{self.url_base}/query', headers=headers, params={
+            'op': 'execute_genquery',
+            'query': f"select DATA_REPL_STATUS where COLL_NAME = '{os.path.dirname(data_object)}' and DATA_NAME = '{os.path.basename(data_object)}'"
+        })
+        #print(r.content) # Debug
+        self.assertEqual(r.status_code, 200)
+
+        result = r.json()
+        self.assertEqual(result['irods_response']['error_code'], 0)
+        self.assertEqual(result['rows'][0][0], '1')
+
+        # Change the replica's status to stale using the modify_replica operation.
+        r = requests.post(self.url_endpoint, headers=headers, data={
+            'op': 'modify_replica',
+            'lpath': data_object,
+            'replica-number': 0,
+            'property-name': 'DATA_REPL_STATUS',
+            'property-value': 0
+        })
+        #print(r.content) # Debug
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(r.json()['irods_response']['error_code'], 0)
+
+        # Show the replica is now marked stale.
+        r = requests.get(f'{self.url_base}/query', headers=headers, params={
+            'op': 'execute_genquery',
+            'query': f"select DATA_REPL_STATUS where COLL_NAME = '{os.path.dirname(data_object)}' and DATA_NAME = '{os.path.basename(data_object)}'"
+        })
+        #print(r.content) # Debug
+        self.assertEqual(r.status_code, 200)
+
+        result = r.json()
+        self.assertEqual(result['irods_response']['error_code'], 0)
+        self.assertEqual(result['rows'][0][0], '0')
+
+        # Remove the data object.
+        r = requests.post(self.url_endpoint, headers=headers, data={
+            'op': 'remove',
+            'lpath': data_object,
+            'no-trash': 1
+        })
```

Comment:
Any reason this isn't in a `finally` block?

Reply:
It's because the testing is still a WIP. Making everything clean is on the TODO list, but that is going to happen at a later time.